api/test/index.js

```diff
@@ -117,3 +117,120 @@ describe('PUT /people/:id', () => {
+                oneEventLink = res.body._links['osdi:events'][0].href
+                expect(res.body).to.have.deep.property('_embedded.osdi:events')
+                done()
+            })
+    })
+
+
+    it('should paginate', done => {
+        request
+            .get(nextLink)
+            .end((err, res) => {
+                expect(res.status).to.equal(200)
+                expect(res.body).to.be.an.object
+                expect(res.body).to.have.deep.property('_links.next')
+                expect(res.body).to.have.deep.property('_links.prev')
+                expect(res.body).to.have.deep.property('_links.osdi:events')
+                expect(res.body).to.have.deep.property('_embedded.osdi:events')
+                done()
+            })
+    })
+
+    it('should accept odata style queries', done => {
+        request
+            .get(nextLink)
+            .end((err, res) => {
+                expect(res.status).to.equal(200)
+                expect(res.body).to.be.an.object
+                expect(res.body).to.have.deep.property('_links.next')
+                expect(res.body).to.have.deep.property('_links.prev')
+                expect(res.body).to.have.deep.property('_links.osdi:events')
+                expect(res.body).to.have.deep.property('_embedded.osdi:events')
+                done()
+            })
+    })
+})
+
+describe('GET /events/:id', () => {
+    it('should return valid hal', done => {
+        request
+            .get(config.baseUrl + '/events?')
+            .end((err, res) => {
+                expect(res.status).to.equal(200)
+                expect(res.body).to.be.an.object
+                done()
+            })
+    })
+})
+
+let newEventId
+describe('POST /events', () => {
+    it('should return the posted person', done => {
+        request
+            .post(config.baseUrl + '/events')
+            .send(require('./person'))
+            .end((err, res) => {
+                expect(res.status).to.equal(200)
+                expect(res.body).to.be.an.object
+                newEventId = res.body.uuid
+                done()
+            })
+    })
+
+    it('person should now be getable', done => {
+        request
+            .get(config.baseUrl + '/events/' + newEventId)
+            .end((err, res) => {
+                expect(res.status).to.equal(200)
+                expect(res.body).to.be.an.object
+                done()
+            })
+    })
+
+    // TODO -- add validation for person and cause the api call to fail
+})
+
+describe('PUT /events/:id', () => {
+    it('should overwrite the property', done => {
+        request
+            .put(config.baseUrl + '/events/' + newEventId)
+            .send({name: 'johnny'})
+            .end((err, res) => {
+                expect(res.status).to.equal(200)
+                expect(res.body.name).to.equal('johnny')
+                done()
+            })
+    })
+
+    it('should not succeed for an incorrect field', done => {
```

Comment:
not sure if this is correct behavior - a request continues even if it has invalid fields, always updates no matter what (just doesn't update fields which aren't in the db obviously)